config/example-menu.cfg

```diff
@@ -3,153 +3,116 @@
+#leave_gcode:
+#enter_gcode:
+#   Script to run on entering to list and leaving from list.
+#   Evaluated as a template.
+
+#[menu some_list some_command]
+#type: command
+#name:
 #enable:
-#parameter:
-#transform:
+#cursor:
+#scroll:
 #gcode:
-#   When menu element is clicked then gcodes on this attribute will be executed.
-#   Can have multiline gcode script and supports output formatting for parameter and transform values.
-#action:
-#   Special action can be executed. Supports [back, exit] menu commands
-#   and [respond response_info] command. Respond command will send '// response_info' to host.
+#   Script to run on button click or long click. Evaluated as a template.
 
-#[menu input1]
+#[menu some_list some_input]
 #type: input
 #name:
-#cursor:
-#width:
 #enable:
-#transform:
-#parameter:
-#   Value from parameter (always index 0) is taken as input value when in edit mode.
-#gcode:
-#   This will be triggered in realtime mode, on exit from edit mode
-#   or in edit mode this will be triggered after click button long press (>0.8sec).
-#longpress_gcode:
-#   In edit mode this will be triggered after click button long press (>0.8sec).
-#   The default is empty. This parameter is optional.
+#cursor:
 #reverse:
-#   This attribute accepts static boolean value.
 #   When enabled it will reverse increment and decrement directions for input.
 #   The default is False. This parameter is optional.
-#readonly:
-#   This attribute accepts same logical expression as 'enable'.
-#   When true then input element is readonly like 'item' and cannot enter to edit mode.
-#   The default is False. This parameter is optional.
-#realtime:
-#   This attribute accepts static boolean value.
-#   When enabled it will execute gcode after each value change.
-#   The default is False. This parameter is optional.
+#input:
+#   Initial value to use when editing - evaluated as a template.
+#   Result must be float.
 #input_min:
-#   It accepts integer or float value. Will set minimal bound for edit value.
-#   The default is 2.2250738585072014e-308. This parameter is optional.
+#   Minimum value of range - evaluated as a template.  Default -99999.
 #input_max:
-#   It accepts integer or float value. Will set maximal bound for edit value.
-#   The default is 1.7976931348623157e+308. This parameter is optional.
+#   Maximum value of range - evaluated as a template.  Default 99999.
 #input_step:
-#   This is mandatory attribute for input.
-#   It accepts positive integer or float value. Will determine increment
-#   and decrement steps for edit value.
+#   Editing step - Must be a positive integer or float value.
 #input_step2:
-#   This is optional attribute for input.
-#   It accepts positive integer or float value. Will determine fast rate
-#   increment and decrement steps for edit value.
+#   Editing step for fast rate - Must be a positive integer or float value.
 #   The default is 0 (input_step will be used instead)
```

Comment:
FWIW, I'd remove `input_step2`.  I think a simple heuristic would be better.  For example, the code could check if `(input_max - input_min) / input_step > 100` and then enable an `internal_input_step2 = 10 * input_step`.